harness/go/history/fetcher.go

```diff
@@ -0,0 +1,139 @@
+package history
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/filter/v1"
+	"go.temporal.io/api/history/v1"
+	"go.temporal.io/api/workflow/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+)
+
+// Fetcher fetches histories.
+type Fetcher struct {
+	Client    client.Client
+	Namespace string
+	TaskQueue string
+	// Approximate value, is given leeway
+	FeatureStarted time.Time
+}
+
+// Fetch returns histories sorted via Histories.Sort. This will continually try
+// to find at least one completed workflow or fail.
+func (f *Fetcher) Fetch(ctx context.Context) (Histories, error) {
```

Comment:
Why do this rather than fetching by run id?

Reply:
> Or is this just grabbing everything after running multiple features to then compare histories later?

Close but not exactly. This collects everything on the task queue after running one feature. But what if one feature runs 5 workflows? That was what was in my head when deciding to "get all histories of everything on the task queue" instead of trying to pluck individual runs.

Also note, this runs for Java and other langs too. I am not requiring that they pass me back the run IDs they created (and hopefully I can avoid a pipe w/ information back from other-lang runner). Rather I give them a task queue to use and let them do whatever.